backend/fms_core/template_importer/row_handlers/normalization/normalization.py

```diff
@@ -0,0 +1,91 @@
+from fms_core.template_importer.row_handlers._generic import GenericRowHandler
+
+from fms_core.models import ProcessMeasurement
+
+from fms_core.services.container import get_container, get_or_create_container
+from fms_core.services.sample import get_sample_from_container, transfer_sample, update_sample
+from fms_core.services.property_value import create_process_measurement_properties
+
+from fms_core.utils import check_truth_like, convert_concentration_from_nm_to_ngbyul
+
+
+class NormalizationRowHandler(GenericRowHandler):
+
+    def process_row_inner(self, source_sample, destination_sample, process_measurement, process_measurement_properties):
+        source_sample_obj, self.errors['sample'], self.warnings['sample'] = get_sample_from_container(
+            barcode=source_sample['container']['barcode'],
+            coordinates=source_sample['coordinates'])
+
+        destination_container_dict = destination_sample['container']
+
+        parent_barcode = destination_container_dict['parent_barcode']
+        if parent_barcode:
+            container_parent, self.errors['parent_container'], self.warnings['parent_container'] = get_container(
+                barcode=parent_barcode)
+        else:
+            container_parent = None
+
+        if source_sample and ((parent_barcode and container_parent) or not parent_barcode):
+            destination_container, _, self.errors['container'], self.warnings['container'] = get_or_create_container(
+                barcode=destination_container_dict['barcode'],
+                kind=destination_container_dict['kind'],
+                name=destination_container_dict['name'],
+                coordinates=destination_container_dict['coordinates'],
+                container_parent=container_parent)
+
+            source_depleted = check_truth_like(source_sample['depleted']) if source_sample['depleted'] else None
```

Comment:
I'm unsure if we should keep the source_depleted conversion here instead of in the importer. It does not really change the performance or readability but it is more of a logical organization. I think of the importer as the place to do simple preprocessing and normalization in order to have more or less final values ready to be passed along. It's not that important just something to think about.